ray-project/kuberay
# [WIP] Added IngressOptions for specifying multiple ingresses/routes

## Why are these changes needed?
Ingresses are created when the `enableIngress` bool is true but this offers no customisation for users who have specific needs with their ingresses/routes.


Closes: #1436 


## Verification Steps
Build the KubeRay image and install the CRDS see this [guide](https://github.com/ray-project/kuberay/blob/master/ray-operator/DEVELOPMENT.md#end-to-end-local-development-process-on-kind)
Create a Ray Cluster with `enableIngress: true` and ingress options in the `headGroupSpec` (see example below)
On creation there should be 2 ingresses/routes with the options you have specified. 
Required fields are `ingressName` and `port`.

```
headGroupSpec:
            enableIngress: true
            ingressOptions:
              ingresses:
              - annotations:
                  <example-annotations>
                host: <example-host>
                ingressClassName: <example-class-name>
                ingressName: <example-ingress-name>
                path: <example-path>
                pathType: <example-path-type>
                port: <example-port>
               - annotations:
                  <example-annotations>
                host: <example-host>
                ingressClassName: <example-class-name>
                ingressName: <example-ingress-name>
                path: <example-path>
                pathType: <example-path-type>
                port: <example-port>
```
Routes only make use of the `annotations`,  `ingressName` and `port`, `tls`, `host` options.
This is the typical route setup for ingressOptions:

```
headGroupSpec:
            enableIngress: true
            ingressOptions:
              ingresses:
              - annotations:
                  <example-annotations>
                host: <example-host>
                ingressName: <example-ingress-name>
                port: <example-port>
                tlsConfig:
	           termination: passthrough
	           insecureEdgePolicy: None
               - annotations:
                  <example-annotations>
                host: <example-host>
                ingressName: <example-ingress-name>
                port: <example-port>
```
## Checks

- Testing Strategy
